### PR TITLE
Fix parent close policy

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**
Split large number of child workflows into small batches to apply parent close policy

**Why?**
To avoid hitting the blob size limit of signal with start workflow API

**How did you test it?**
unit tests

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/history/task/transfer_active_task_executor.go [hunk 1]
```diff
@@ -1790,17 +1793,27 @@ func (t *transferActiveTaskExecutor) processParentClosePolicy(
...
 				Policy:     childInfo.ParentClosePolicy,
 			})
+
+			if len(executions) == batchSize {
+				err := t.parentClosePolicyClient.SendParentClosePolicyRequest(ctx, parentclosepolicy.Request{
+					DomainName: domainName,
+					Executions: executions,
+				})
+				if err != nil {
+					return err
```
Comment (taylanisikdemir) [inline service/history/task/transfer_active_task_executor.go:1803]: retry attempt of the task will have to go thru each batch. is this operation idempotent?

File: service/history/task/transfer_active_task_executor.go [hunk 2]
```diff
@@ -1777,8 +1777,11 @@ func (t *transferActiveTaskExecutor) processParentClosePolicy(
 	if t.shard.GetConfig().EnableParentClosePolicyWorker() &&
 		len(childInfos) >= t.shard.GetConfig().ParentClosePolicyThreshold(domainName) {
 
-		executions := make([]parentclosepolicy.RequestDetail, 0, len(childInfos))
+		batchSize := t.shard.GetConfig().ParentClosePolicyBatchSize(domainName)
+		executions := make([]parentclosepolicy.RequestDetail, 0, common.MinInt(len(childInfos), batchSize))
+		count := 0
```
Comment (taylanisikdemir) [inline service/history/task/transfer_active_task_executor.go:1782]: nit: use loop index variable instead

**Review by taylanisikdemir — APPROVED** [review]

File: service/history/task/transfer_active_task_executor.go (hunk 2 above)
Comment (Shaddoll) [reply service/history/task/transfer_active_task_executor.go:1782]: sadly, childInfos is a map, not a slice

File: service/history/task/transfer_active_task_executor.go (hunk 1 above)
Comment (Shaddoll) [reply service/history/task/transfer_active_task_executor.go:1803]: the operation signal a workflow and it is idempotent

**Pull request merged**